runner/operator/access/__init__.py

```diff
@@ -35,21 +35,50 @@ def get_request_id_runs(request_id):
     return request_id_runs
 
 
-def get_unfiltered_matched_normal(patient_id):
+def get_unfiltered_matched_normal(patient_id, request_id=None):
     """
     Find a matched normal sample for the given patient ID with the following precedence:
 
-    1. Latest Matched Normal from IGO ACCESS samples
-    2. Latest Matched Normal from DMP ACCESS samples
-    3. Return (None, ''), which will be used as a placeholder for skipping genotyping in the SNV pipeline
+    1. Latest Matched Normal from IGO ACCESS samples (same request)
+    2. Latest Matched Normal from IGO ACCESS samples (any request)
+    3. Latest Matched Normal from DMP ACCESS samples
+    4. Return (None, ''), which will be used as a placeholder for skipping genotyping in the SNV pipeline
 
     Todo: generalize to work for duplex / simplex / standard, and use in MSI operator
 
+    :param: patient_id - str Patient ID in CMO format (C-ABC123)
+    :param: request_id - str IGO request ID (06302_AA)
     :return: (file_system.models.File - bam, str - sample_id)
     """
     patient_normals_search = patient_id + NORMAL_SAMPLE_SEARCH
-    unfiltered_matched_normal_bam = File.objects.filter(file_name__startswith=patient_normals_search, file_name__endswith=IGO_UNFILTERED_REGEX)
-    if len(unfiltered_matched_normal_bam) == 0:
+    unfiltered_matched_normal_bam = None
+
+    # Case 1
+    if request_id:
+        unfiltered_matched_normal_bam = File.objects.filter(
+            file_name__startswith=patient_normals_search,
+            file_name__endswith=IGO_UNFILTERED_REGEX,
+            port__run__tags__request_id__startswith=request_id.split('_')[0]
+        )
+    # Case 2
+    if not request_id or len(unfiltered_matched_normal_bam) == 0:
+        unfiltered_matched_normal_bam = File.objects.filter(
+            file_name__startswith=patient_normals_search,
+            file_name__endswith=IGO_UNFILTERED_REGEX
+        )
+
+    if len(unfiltered_matched_normal_bam) > 1:
+        msg = 'WARNING: Found more than one matching unfiltered normal bam file for patient {}. ' \
+              'We will choose the most recently-created one for this run.'
+        msg = msg.format(patient_id)
+        logger.warning(msg)
```

Comment:
I think for debugging it'd help to have the group_id in addition to the patient id, you can access it with `self.job_group_id`

Reply:
because this is an external function I don't have access to that field of the operator, it seems a bit messy to pass that in to this function but is there another way to get it?